python/src/aac/lang/language_context.py

```diff
@@ -189,6 +261,21 @@ def does_definition_root_match(definition: Definition) -> bool:
 
         return list(filter(does_definition_root_match, self.definitions))
 
+    def get_definitions_by_file_uri(self, file_uri: str) -> list[Definition]:
+        """Return a subset of definitions that are sourced from the target file URI.
+
+        Args:
+            file_uri (str): The source file URI to filter on.
+
+        Returns:
+            A list of definitions belonging to the target file.
+        """
+
+        def does_definition_source_uri_match(definition: Definition) -> bool:
+            return file_uri == definition.lexemes[0].source
```

Comment:
I've identified some blockers in the current way of linking to definition source's in this issue: https://github.com/jondavid-black/AaC/issues/322